core/src/main/scala/org/apache/spark/sql/store/CodeGeneration.scala

```diff
@@ -260,6 +283,32 @@ object CodeGeneration extends Logging {
         "dialect")).asInstanceOf[GeneratedStatement]
   }
 
+
+  private[this] def compileGeneratedIndexUpdate(table: String,
```

Comment:
I am not sure what additional complexity you are talking about, it is reusing table generation code refactored.